sotodlib/site_pipeline/analyze_bright_ptsrc.py

```diff
@@ -63,71 +48,77 @@ def get_xieta_src_centered(ctime, az, el, data, sso_name, threshold=5):
 
     csl = so3g.proj.CelestialSightLine.az_el(ctime, az, el, weather="typical")
     q_bore = csl.Q
-    
-    ### Signal strength criterion - a chance to flag detectors (should be extra rare)
+
+    # Signal strength criterion - a chance to flag detectors (should be extra
+    # rare)
     peaks, _ = scipy.signal.find_peaks(data, height=threshold * np.std(data))
     if not list(peaks):
         return
 
-    ## The mean planet position is used
-    dt = datetime.fromtimestamp(np.mean(ctime))
-    dt_str = dt.strftime("%Y/%m/%d %H:%M:%S")
-    obj = getattr(ephem, sso_name)()
-    obj.compute(dt_str)
-    ra_obj = obj.ra
-    dec_obj = obj.dec
-    q_obj = quat.rotation_lonlat(ra_obj, dec_obj)
-
-    ### rotate coord system around max point
+    # Need the planet position at the peak
     src_idx = np.where(data == np.max(data[peaks]))[0][0]
-    q_bs_t = coords.quat.quat(*np.array(q_bore)[src_idx])
-    xi0, eta0, _ = quat.decompose_xieta(~q_bs_t * q_obj)
-    q_det = quat.rotation_xieta(xi0, eta0)
-    q_total = ~q_det * ~q_bore * q_obj
+    d1_unix = ctime[src_idx]
+    planet = planets.SlowSource.for_named_source(sso_name, d1_unix * 1.)
+    ra0, dec0 = planet.pos(d1_unix)
+    q_obj = so3g.proj.quat.rotation_lonlat(ra0, dec0)
+
+    # rotate coord system around max point
+    q_bs_t = coords.quat.quat(*np.array(~q_bore * q_obj)[src_idx])
+    q_total = ~q_bs_t * ~q_bore * q_obj
 
     xi, eta, _ = quat.decompose_xieta(q_total)
-    xi0, eta0, _ = quat.decompose_xieta(q_det)
+    xi0, eta0, _ = quat.decompose_xieta(q_bs_t)
 
-    return np.array([xi, eta]), np.array([xi0, eta0]), q_det, q_bore, q_obj
+    return np.array([xi, eta]), np.array([xi0, eta0]), q_bs_t, q_bore, q_obj
 
 
-def define_fit_params(band, tele, return_beamsize=False):
+def define_fit_params(config_file_path, tele, tube, sso_name):
     """
-    Define initial position and allowed offsets for detector centroids
-    and beam parameters according to frequency and telescope specifications.
-    The time constant initial guess and boundary conditions remain the same.
+    Define initial position and allowed offsets for detector centroids,
+    beam parameters and time constant as read from configuration file
+    located at 'config_file_path'.
```

Comment:
Oh, the example file is the same as before; I must have removed this from the current script to avoid hardcoded paths. It can be found at https://github.com/simonsobs/pwg-scripts/blob/master/pwg-bcp/sotodlib_staging/data/initial_parameters_fit_pt_src.yaml. Sorry about that. I can also write the documentation; there is no problem. I avoided doing it at the time of the PR because I was in a rush to leave for a plane, I think.